scanner/metadata/metadata.go

```diff
@@ -105,8 +107,47 @@ func (t Tags) OriginalDate() (int, string) { return t.getDate("originaldate") }
 func (t Tags) ReleaseDate() (int, string)  { return t.getDate("releasedate") }
 func (t Tags) Comment() string             { return t.getFirstTagValue("comment") }
 func (t Tags) Lyrics() string {
-	return t.getFirstTagValue("lyrics", "lyrics-eng", "unsynced_lyrics", "unsynced lyrics", "unsyncedlyrics")
+	lyrics := model.Lyrics{}
+	basicLyrics := t.getAllTagValues("lyrics", "unsynced_lyrics", "unsynced lyrics", "unsyncedlyrics")
+
+	for _, value := range basicLyrics {
+		lyric, err := model.ToLyrics("xxx", value)
+		if err != nil {
+			log.Warn("Unexpected failure occurred when parsing lyrics", "file", t.filePath, "error", err)
+			continue
+		}
+
+		lyrics = append(lyrics, *lyric)
+	}
+
+	for tag, value := range t.tags {
+		if strings.HasPrefix(tag, "lyrics-") {
+			language := strings.TrimSpace(strings.TrimPrefix(tag, "lyrics-"))
+
+			if language == "" {
+				language = "xxx"
+			}
+
+			for _, text := range value {
+				lyric, err := model.ToLyrics(language, text)
+				if err != nil {
+					log.Warn("Unexpected failure occurred when parsing lyrics", "file", t.filePath, "error", err)
+					continue
+				}
+
+				lyrics = append(lyrics, *lyric)
+			}
+		}
+	}
+
+	res, err := json.Marshal(lyrics)
+	if err != nil {
+		log.Warn("Unexpected error occurred when serializing lyrics", "file", t.filePath, "error", err)
+		return ""
+	}
+	return string(res[:])
```

Comment:
Never saw this syntax before. Why not simply use `string(res)`

Reply:
Ah, ok. I actually used this before, but this is only required for converting arrays (fixed length) to slices (variable lengths). In this case it is not necessary. There are other instances of this in your PR, please remove if you find it.